<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">

    <xacro:macro name="canopen_ros2_control_test" params="
      name
      prefix
      bus_config
      master_config
      can_interface_name
      master_bin">

        <ros2_control name="${name}" type="system">
            <hardware>
              <plugin>canopen_ros2_control/CanopenSystem</plugin>
              <param name="bus_config">${bus_config}</param>
              <param name="master_config">${master_config}</param>
              <param name="can_interface_name">${can_interface_name}</param>
              <param name="master_bin">"${master_bin}"</param>
            </hardware>
            <joint name="${prefix}node_1">
                <param name="node_id">2</param>
                <command_interface name="position">
                  <param name="min">{-2*pi}</param>
                  <param name="max">{2*pi}</param>
                </command_interface>
                <!--<command_interface name="velocity">-->
                  <!--<param name="min">-3.15</param>-->
                  <!--<param name="max">3.15</param>-->
                <!--</command_interface>-->
                <state_interface name="position"/>
                <!--<state_interface name="velocity"/>-->
                <!--<state_interface name="effort"/>-->
            </joint>
        </ros2_control>

    </xacro:macro>

</robot>